beacon-chain/slasher/detect_attestations.go

```diff
@@ -138,17 +138,32 @@ func (s *Service) checkSurroundVotes(
 	}
 
 	// Save the updated chunks to disk.
-	if err := s.saveChunksToDisk(ctx, slashertypes.MinSpan, minChunkByChunkIndexByValidatorChunkIndex); err != nil {
+	if err := s.saveChunksToDisk(ctx, slashertypes.MinSpan, filterByUpdatedChunk(minChunkByChunkIndexByValidatorChunkIndex)); err != nil {
 		return nil, errors.Wrap(err, "could not save updated min chunks to disk")
 	}
 
-	if err := s.saveChunksToDisk(ctx, slashertypes.MaxSpan, maxChunkByChunkIndexByValidatorChunkIndex); err != nil {
+	if err := s.saveChunksToDisk(ctx, slashertypes.MaxSpan, filterByUpdatedChunk(maxChunkByChunkIndexByValidatorChunkIndex)); err != nil {
 		return nil, errors.Wrap(err, "could not save updated max chunks to disk")
 	}
 
```

Comment:
I need to point out that the property ```hasBeenUpdated``` should be set to ```false``` here as we are saving the chunks onto the disk. However, doing so with the current implementation would require adding a ```SetHasBeenMutated``` function to the interface, which, considering immutability, is not ideal.

This part of the slasher should be refactored so it returns a map which allows this function to know exactly which chunks should be stored onto the disk.

I will open an issue to further improve the performance of the slasher if that's okay with the team.